packages/react-utilities/src/compose/types.ts

```diff
@@ -29,17 +29,32 @@ export type DefaultObjectShorthandProps = ObjectShorthandProps<
 >;
 
 /**
- * Defines the slot props for a slot that supports a Component type.
- *
- * For intrinsic elements like 'div', use {@link IntrinsicShorthandProps} instead.
+ * Takes the props we want to support for a slot and adds the ability for `children` to be a render function that takes
+ * those props.
  */
 export type ObjectShorthandProps<Props extends { children?: React.ReactNode } = {}> = Props & {
   children?: Props['children'] | ShorthandRenderFunction<Props>;
 };
 
+/**
+ * Defines the slot props for a slot that supports a Component type.
+ *
+ * For intrinsic elements like 'div', use {@link IntrinsicShorthandProps} instead.
+ *
+ * The template param is the type of a control, using the `typeof` keyword. For example:
```

Comment:
```suggestion
 * The generic param is a type of a react component. 
 *
 * @example
 * ```
 * interface Props {....} 
 * const Button: React.FC<Props> = () => {...}
 * // $ExpectType ...
 * type  SlotProps = ComponentSlotProps<typeof Button>
 * ```
```

I'm not sure I understand what this mapped type would return from perspective of new user. 

it would be nice if that could get documented (see my suggestion - meaning fill out the interface and what would be the `$ExpectType`)

I think public facing apis should be well documented, WDYT ?